File: frontend/src/types/calendar_journal.ts
import { z } from "zod";

export const CalendarDaySchema = z.object({
  date: z.string(),
  in_month: z.boolean(),
  realized_pnl: z.number(),
  trade_count: z.number().int(),
  trade_ids: z.array(z.number().int()).default([]),
  is_today: z.boolean().default(false),
});
export type CalendarDay = z.infer<typeof CalendarDaySchema>;

export const CalendarWeekSchema = z.object({
  week_of_month: z.number().int(),
  days: z.array(CalendarDaySchema),
  realized_pnl: z.number(),
  trade_count: z.number().int(),
});
export type CalendarWeek = z.infer<typeof CalendarWeekSchema>;

export const CalendarMonthSchema = z.object({
  month: z.string(),       // "YYYY-MM"
  label: z.string(),       // "May 2026"
  weeks: z.array(CalendarWeekSchema),
  realized_pnl: z.number(),
  trade_count: z.number().int(),
});
export type CalendarMonth = z.infer<typeof CalendarMonthSchema>;

File: backend/schemas/calendar_journal.py
"""API schema for the journal P&L calendar — mirrors the dataclasses
in calculations/journal_calendar.py."""

from __future__ import annotations

from pydantic import BaseModel, Field


class CalendarDayOut(BaseModel):
    date: str
    in_month: bool
    realized_pnl: float
    trade_count: int
    trade_ids: list[int] = Field(default_factory=list)
    is_today: bool = False


class CalendarWeekOut(BaseModel):
    week_of_month: int
    days: list[CalendarDayOut]
    realized_pnl: float
    trade_count: int


class CalendarMonthOut(BaseModel):
    month: str
    label: str
    weeks: list[CalendarWeekOut]
    realized_pnl: float
    trade_count: int
